redis/redispipebp/config.go

```diff
@@ -0,0 +1,202 @@
+package redispipebp
+
+import (
+	"time"
+
+	"github.com/joomcode/redispipe/rediscluster"
+	"github.com/joomcode/redispipe/redisconn"
+)
+
+// ClientConfig can be used to configure a redispipe "Client".  See the docs for
+// redisconn.Opts in redispipe for details on what each value means and what its
+// defaults are:
+// https://pkg.go.dev/github.com/joomcode/redispipe@v0.9.4/redisconn?tab=doc#Opts
+//
+// Can be deserialized from YAML.
+//
+// Examples:
+//
+// Minimal YAML:
+//
+//	redis:
+//	 addr: localhost:6379
+//
+// Full YAML:
+//
+//	redis:
+//	 addr: redis://localhost:6379
+//	 opts:
+//	  db: 1
+//	  writePause: 150000ns # 150 microseconds
+//	  reconnectPause: 2s
+//	  tcpKeepAlive: 333ms
+//	  asyncDial: false
+//	  scriptMode: false
+//	  timeouts:
+//	   dial: 1s
+//	   io: 500ms
+type ClientConfig struct {
+	// Addr is the address of your Redis server.
+	//
+	// This is the only required field.
+	Addr string `yaml:"addr"`
+
+	Options Opts `yaml:"options"`
+}
+
+// Opts returns a redisconn.Opts object with the configured options applied.
+func (cfg ClientConfig) Opts() redisconn.Opts {
+	opts := redisconn.Opts{}
+	cfg.Options.ApplyOpts(&opts)
+	return opts
+}
+
+// ClusterConfig can be used to configure a redispipe "Cluster".  See the docs for
+// redisluster.Opts in redispipe.
+//
+// Can be deserialized from YAML.
+//
+// Examples:
+//
+// Minimal YAML:
+//
+//	redis:
+//	 addrs:
+//	  - localhost:6379
+//	  - localhost:6380
+//
+// Full YAML:
+//
+//	redis:
+//	 addrs:
+//	  - localhost:6379
+//	  - localhost:6380
+//	 options:
+//	  writePause: 150000ns # 150 microseconds
+//	  reconnectPause: 2s
+//	  tcpKeepAlive: 333ms
+//	  asyncDial: false
+//	  scriptMode: false
+//	  timeouts:
+//	   dial: 1s
+//	   io: 500ms
+type ClusterConfig struct {
+	// Addrs is the list of seed addresses to connect to the Redis Cluster.
+	Addrs []string `yaml:"addrs"`
+
+	// Options maps to rediscluster.Opts.HostOpts
+	Options Opts `yaml:"options"`
+
+	// Cluster is the remaining, Redis Cluster specific options.
+	Cluster ClusterOpts `yaml:"cluster"`
+}
+
+// Opts returns a rediscluster.Opts object with the configured options applied.
+func (cfg ClusterConfig) Opts() rediscluster.Opts {
+	opts := rediscluster.Opts{}
+	cfg.Options.ApplyOpts(&opts.HostOpts)
+	cfg.Cluster.ApplyOpts(&opts)
+	return opts
+}
+
+// Opts is used to apply the configured options to a rediscluster.Opts.
+//
+// All fields in Opts are optional and redispipe provides default values.
+//
+// See the documentation for details on each field as well as the default values.
+// https://pkg.go.dev/github.com/joomcode/redispipe@v0.9.4/redisconn?tab=doc#Opts
+type Opts struct {
+	// DB is the database number in Redis.
+	//
+	// Maps to redisconn.Opts.DB
+	DB *int `yaml:"db"`
+
+	// WritePause is the duration of time the write loop should wait to collect
+	// commands to flush.
+	//
+	// Maps to redisconn.Opts.WritePause
+	WritePause *time.Duration `yaml:"writePause"`
```

Comment:
🔕 Maybe for durations, using non-pointers and checking for `>0` instead of `!= nil` instead is better?